cloud/deployment/deployment_test.go

```diff
@@ -1065,10 +1068,10 @@ func (s *Suite) TestCreate() {
 
 	s.Run("success with standard/dedicated type different scheduler sizes", func() {
 		// Set up mock responses and expectations
-		mockCoreClient.On("GetDeploymentOptionsWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&GetDeploymentOptionsResponseOK, nil).Times(6)
-		mockCoreClient.On("ListWorkspacesWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&ListWorkspacesResponseOK, nil).Times(6)
-		mockPlatformCoreClient.On("CreateDeploymentWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&mockCreateDeploymentResponse, nil).Times(6)
-		mockPlatformCoreClient.On("ListClustersWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&mockListClustersResponse, nil).Times(3)
+		mockCoreClient.On("GetDeploymentOptionsWithResponse", mock.Anything, mock.Anything, mock.Anything).Return(&GetDeploymentOptionsResponseOK, nil).Times(8)
```

Comment:
Not strictly necessary in this PR but would love to see us start to convert these `.Times(n)` tests into table-driven tests.

Reply:
Yeah 💯 , also writing the tests made me realize there is a scope of making them stateless, as some of the states are being passed between the tests, which makes it very hard to add/update any tests. Let me drop an issue on this to refactor `deployment/deployment_test.go` to use table-driven tests and test suites, and I will get to it in my free time.

Reply:
Add a separate issue on our board: https://github.com/astronomer/astro-cli/issues/1698